raiden/blockchain_events_handler.py

```diff
@@ -226,38 +237,90 @@ def on_blockchain_event(raiden, event):
+    if data['event'] == EVENT_TOKEN_ADDED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['channel_manager_address'] = address_decoder(data['args']['channel_manager_address'])
         handle_tokennetwork_new(raiden, event)
 
-    elif data['event'] == 'ChannelNew':
+    elif data['event'] == EVENT_CHANNEL_NEW:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['participant1'] = address_decoder(data['args']['participant1'])
         data['participant2'] = address_decoder(data['args']['participant2'])
         handle_channel_new(raiden, event)
 
-    elif data['event'] == 'ChannelNewBalance':
+    elif data['event'] == EVENT_CHANNEL_NEW_BALANCE:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['token_address'] = address_decoder(data['args']['token_address'])
         data['participant'] = address_decoder(data['args']['participant'])
         data['balance'] = data['args']['balance']
         handle_channel_new_balance(raiden, event)
 
-    elif data['event'] == 'ChannelClosed':
+    elif data['event'] == EVENT_CHANNEL_CLOSED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['closing_address'] = address_decoder(data['args']['closing_address'])
         handle_channel_closed(raiden, event)
 
-    elif data['event'] == 'ChannelSettled':
+    elif data['event'] == EVENT_CHANNEL_SETTLED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         handle_channel_settled(raiden, event)
 
-    elif data['event'] == 'ChannelSecretRevealed':
+    elif data['event'] == EVENT_CHANNEL_SECRET_REVEALED:
         data['registry_address'] = address_decoder(data['args']['registry_address'])
         data['receiver_address'] = address_decoder(data['args']['receiver_address'])
         data['secret'] = data['args']['secret']
         handle_channel_withdraw(raiden, event)
 
     else:
         log.error('Unknown event type', event=event)
+
+
+def on_blockchain_event2(raiden, event):
+    log.debug('EVENT', node=pex(raiden.address), chain_event=event)
+
+    event = decode_event_to_internal(event)
+    data = event.event_data
+
+    if data['args'].get('channel_identifier'):
+        data['channel_identifier'] = data['args'].get('channel_identifier')
+
+    if data['event'] == EVENT_TOKEN_ADDED:
+        handle_tokennetwork_new(raiden, event)
+
+    elif data['event'] == EVENT_CHANNEL_NEW:
+        data['settle_timeout'] = data['args']['settle_timeout']
+        handle_channel_new(raiden, event)
+
+    elif data['event'] == EVENT_CHANNEL_NEW_BALANCE:
+        data['deposit'] = data['args']['deposit']
+        handle_channel_new_balance(raiden, event)
+
+    elif data['event'] == EVENT_CHANNEL_WITHDRAW:
+        data['withdrawn_amount'] = data['args']['withdrawn_amount']
+        # handle_channel_withdraw(raiden, event)
```

Comment:
This is a comment, because the `handle_channel_withdraw` existent now is badly named - it actually does unlocking. We should use this name for the `setTotalWithdraw` function of the contract.
So, because these functions either don't exist or are not implemented as expected, I just used the comments to mark what will exist here.